OpenRA.Game/GameRules/ActorInfo.cs

```diff
@@ -100,7 +100,7 @@ public IEnumerable<ITraitInfo> TraitsInConstructOrder()
 			var unresolved = source.Except(resolved);
 
 			var testResolve = new Func<Type, Type, bool>((a, b) => a == b || a.IsAssignableFrom(b));
-			var more = unresolved.Where(u => u.Dependencies.All(d => resolved.Exists(r => testResolve(d, r.Type))));
+			var more = unresolved.Where(u => u.Dependencies.All(d => resolved.Exists(r => testResolve(d, r.Type)) && !unresolved.Any(u1 => testResolve(d, u1.Type))));
```

Comment:
Vagely related scope creep (not a blocker): Using LINQ here might be compact, but it makes what is otherwise a simple algorithm really difficult to parse.  How much would I have to bribe to you rewrite this block of 6 lines in a clearer way or to add comments?